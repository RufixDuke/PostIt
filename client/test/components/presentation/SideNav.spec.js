import React from "react";
import { mount } from "enzyme";
import SideNav from '../../../src/components/presentation/SideNav';

describe("SideNav Component", () => {
  let props;
  let mountedSideNav;
  const currentUser = {
    username: 'user1',
    email: 'user1@gmail.com'
  };

  const sideNav = () => {
    if (!mountedSideNav) {
      mountedSideNav = mount(
        <SideNav {...props} />
      );
    }
    return mountedSideNav;
  }

  beforeEach(() => {
    props = {
      currentUser
    };
    mountedSideNav = undefined;
  });

  it("should always render ul", () => {
    const divs = sideNav().find("ul");
    expect(sideNav()).toMatchSnapshot();
    expect(divs.length).toBeGreaterThan(0);
  });

  describe("the rendered div", () => {
    it("should contain everything else that gets rendered", () => {
      const divs = sideNav().find("ul");
      const wrappingDiv = divs.first();
      expect(wrappingDiv.children()).toEqual(sideNav().children());
    });
  });

  describe("rendered SideNav", () => {
    it("should receive props", () => {
      expect(Object.keys(sideNav().props()).length).toBe(1);
    });
  });

  describe("when currentUser prop is defined", () => {
    it("should display the username", () => {
      expect(sideNav().find('.current-username').text()).toBe("user1");
    });
    it("should display the email", () => {
      expect(sideNav().find('.current-email').text()).toBe("user1@gmail.com");
    });
  });
});
