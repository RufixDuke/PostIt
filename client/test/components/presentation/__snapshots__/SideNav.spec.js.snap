// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNav Component should always render ul 1`] = `
<SideNav
  currentUser={
    Object {
      "email": "user1@gmail.com",
      "username": "user1",
    }
  }
>
  <ul
    className="side-nav groupdetails-nav"
    id="slide-out"
  >
    <li>
      <h5
        className="white-text name user-size current-username"
      >
        user1
      </h5>
    </li>
    <li>
      <h5
        className="white-text email user-size current-email"
      >
        user1@gmail.com
      </h5>
    </li>
    <li>
      <Link
        className="waves-effect waves-light green nav-anchor"
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/members"
      >
        <a
          className="waves-effect waves-light green nav-anchor"
          onClick={[Function]}
          style={Object {}}
        >
          <i
            aria-hidden="true"
            className="fa fa-users snav"
          />
          Members
        </a>
      </Link>
    </li>
    <li>
      <div
        className="divider"
      />
    </li>
    <li>
      <a
        className="waves-effect waves-light green nav-anchor modal-trigger"
        href="#addUser"
      >
        <i
          aria-hidden="true"
          className="fa fa-user-plus snav"
        />
        Add User
      </a>
    </li>
  </ul>
</SideNav>
`;
